ocaml/libs/xapi-stdext/lib/xapi-stdext-unix/test/unixext_test.ml

```diff
@@ -187,7 +187,97 @@ let test_proxy =
       expect_string ~expected:write.data ~actual:read.data ;
       true
 
-let tests = [test_proxy; test_time_limited_write; test_time_limited_read]
+let mtime_span_gen =
+  let open Gen in
+  (* We can't use Mtime.Span.max_span, because that is too big for float,
+     only works for int64 conversion, and there is no Mtime.Span.max_span_float.
+     The documentation says that 2^53 is the maximum though, so use that.
+     Otherwise we'll fail later when converting to string and back goes through float.
+
+     Use microseconds instead of nanoseconds, because nanoseconds have rounding
+     errors during conversion.
+  *)
+  let+ usval = 0 -- (((1 lsl 53) - 1) / 1_000_000) in
+  Mtime.Span.(usval * us)
+
+let test_timeout_string_conv =
+  let gen = mtime_span_gen and print = Fmt.to_to_string Mtime.Span.pp in
+  Test.make ~name:__FUNCTION__ ~print gen @@ fun timeout ->
+  let str = Clock.Timer.span_to_s timeout |> Printf.sprintf "%.6f" in
+  let timeout' =
+    str |> Float.of_string |> Clock.Timer.s_to_span |> Option.get
+  in
+  if not (Mtime.Span.equal timeout timeout') then
+    Test.fail_reportf
+      "timeout not equal after round-trip through %S: %Lu (%a) <> %Lu (%a)" str
+      (Mtime.Span.to_uint64_ns (timeout :> Mtime.Span.t))
+      Mtime.Span.pp timeout
+      (Mtime.Span.to_uint64_ns (timeout' :> Mtime.Span.t))
+      Mtime.Span.pp timeout' ;
+  true
+
+let spans =
+  Gen.oneofa ([|1; 100; 300|] |> Array.map (fun v -> Mtime.Span.(v * ms)))
+
+let test_timer_remaining =
```

Comment:
This test is now part of Clock.Timer: https://github.com/xapi-project/xen-api/blob/master/ocaml/libs/clock/test_timer.ml#L84

Reply:
Thanks, I dropped the code from here in a fixup, we can squash that once I rebase on latest master.